root/usr/share/fdi/facts/discovery-facts.rb

```diff
@@ -161,3 +162,46 @@ def add_ipmi_facts fact, value, channel = nil
     end
   end
 end
+
+# NetworkManager details (e.g. nmprimary_dhcp4_option_domain_name)
+nmout = Facter::Util::Resolution.exec("nmcli -t con show primary 2>/dev/null")
+nmout.each_line do |x|
+  elements = x.split(":", 2)
+  name = elements.first.downcase.sub(/\[\d+\]$/,"")
+  if name =~ /dhcp.\.option/
+    dhcp_elems = elements.last.split(/\s*=\s*/, 2)
+    name += '_' + dhcp_elems.first
+    value = dhcp_elems.last
+  else
+    value = elements.last
+  end
+  name = "nmprimary_" + name.tr('.', '_')
+  Facter.add(name) do
+    setcode do
+      value.chomp
+    end
+  end
+end
+
+# Create DHCP FQDN helper facts
+Facter.add("nmprimary_dhcp4_option_fqdn") do
+  setcode do
+    if Facter.value["nmprimary_dhcp4_option_host_name"] && Facter.value["nmprimary_dhcp4_option_domain_name"]
```

Comment:
You need to use round brackets here. `Facter.value()`. `Facter.value[]` raises an error for me.
Same below.